### DISCUSSION TITLE
How to pass data to frontend?

### DISCUSSION THREAD

**Discussion by Manas-Nagelia** [Mathieu2301/TradingView-API#195] [Q&A] [answered]
Hello. 

I am using Next.js 13 (app directories in beta) and have trouble passing the pricing data from the backend to the client side. I'm having the real-time chart data run in a server component, but I'm having trouble passing the data from the `onUpdate` function to the front-end React component. Adding a `return` statement to the `onUpdate` function does not work, and moving the code to the client side (React) does not work either. 

I'm planning just to store the real-time data in a database to combat this issue, but are there any other ways to combat this issue, not relying on a database? 

Thanks.

**Comment by Mathieu2301** [comment, marked as answer]
This question is not related to this GitHub repository but to NextJS. You should use websockets to pass realtime data.
